File: AppCV-Fiches/WebCV-Fiches/Models/CVViewModels/CVViewModel.cs
using DAL_CV_Fiches.Models;
using DAL_CV_Fiches.Models.Graph;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace WebCV_Fiches.Models.CVViewModels
{
    public class CVViewModel
    {
        public string GraphId { get; set; }

        public string Prenom { get; set; }
        public string Nom { get; set; }
        public string Fonction { get; set; }
        public string Biographie { get; set; }
        public List<DomaineDInterventionViewModel> DomainesDIntervention { get; set; }
        public List<FormationAcademiqueViewModel> FormationsAcademique { get; set; }
        public List<CertificationViewModel> Certifications { get; set; }
        public List<MandatViewModel> Mandats { get; set; }
        public List<TechnologieViewModel> Technologies { get; set; }
        public List<PerfectionnementViewModel> Perfectionnements { get; set; }
        public List<LangueViewModel> Langues { get; set; }
    }
    
}

File: AppCV-FichesFront/src/app/Models/cvview-model.ts
import { DomaineDInterventionViewModel } from "./DomaineDIntervention-model";
import { FormationAcademiqueViewModel } from "./FormationAcademique-model";
import { CertificationViewModel } from "./Certification-model";
import { MandatViewModel } from "./Mandat-model";
import { TechnologieViewModel } from "./Technologie-model";
import { PerfectionnementViewModel } from "./Perfectionnement-model";
import { LangueViewModel } from "./Langue-model";

export class CVViewModel {
  public graphIdConseiller: string;
  public graphIdUtilisateur: string;
  public graphIdFonction: string;
  public graphIdCV: string;

  public prenom: string;
  public nom: string;
  public fonction: string;
  public biographie: string;
  public domainesdintervention: Array<DomaineDInterventionViewModel>;
  public formationsacademique: Array<FormationAcademiqueViewModel>;

  certifications: Array<CertificationViewModel>;
  mandats: Array<MandatViewModel>;
  technologies: Array<TechnologieViewModel>;
  perfectionnements: Array<PerfectionnementViewModel>;
  langues: Array<LangueViewModel>;

  constructor() {
    this.domainesdintervention = new Array<DomaineDInterventionViewModel>();
    this.formationsacademique = new Array<FormationAcademiqueViewModel>();
    this.certifications = new Array<CertificationViewModel>();
    this.mandats = new Array<MandatViewModel>();
    this.technologies = new Array<TechnologieViewModel>();
    this.perfectionnements = new Array<PerfectionnementViewModel>();
    this.langues = new Array<LangueViewModel>();
  }
}
